content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+dynamically loaded, this is called `Code Splitting`, then `module`
+will be an independent file lazy loaded, only and when its needed.
+
+Latest version of Webpack make this configuration almost trivial,
+check out their [docs][110].
+
+This approach also works with the _Classic Way_ of using dependencies via Browser Globals.
+
+## Active vs Passive compilation
+
+> **TLDR: Do not use _Passively compiled_ dependecies ever!**
+
+I have seen situations in which packages are distributed raw,
+without being compiled before being published, this means **Component Authors**
+package and distribute the raw sources like Typescript, ES2054, JSX, CSS Modules, etc.,
+and rely on **Component Consumers**, like Applications, to compile them usually
+by configuring Webpack to also treat `node_modules/my-raw-dependency`
+as part of the Application source.
+
+This is what I call **Passive compilation** because **Component Authors**
+delegate their package compilation to the **Component Consumers**, instead
+of **Active compilation** in which each package is distributed already compiled
+by its **Component Author**.
+
+Today you can go a long way with Webpack abstractions like
+[Create React App][16] and [Create React Library][105], but you need to
+do things in a fairly standard way and **Passive Compilation** is not at all
+a standard way of distributing code.
+
+In my experience Webpack configurations are something that most developers
+do not know or do not want to touch so
+**having these sort of abstractions inside
+your company will save you a lot of trouble and maintenance,
+which translates to less operation costs**.
+These Webpack abstractions have lots of smart people working on them and
+pouring their past experiences in creating something that will cover
+lots of use cases you probably wont ever think of, so you are getting even
+more things for free than you anticipate.
+
+Additionally, having custom Webpack configurations creates a potential
+opportunity to tightly couple other parts of your system like configurations,
+Services URLs, CDNs, PORT assignment, etc. to your Webpack build and configuration
+which makes it potentially brittle to any tweaks since it might affect some
+tightly coupled relationship in some obscure place. Relying on standard Webpack abstractions
+provides a very clear interface where you can pass configuration values and custom behaviors
+effectively reducing this coupling.
+
+Finally, and with a more of a "business" look, it is much more
+efficient and effective to rely on pre-made solutions for Application bundling and compilation.
+Webpack and the other bundlers are hard to configure exactly right and as a company you will
+need to maintain and bug fix your hand made configurations if you do them yourself.
+It is effectively one more source of bugs you will have to deal with,
+so rely as much as possible on third party open source solutions and for that
+you _must_ avoid **Passive Compilation**.
```

Comment:
Without knowing much about Webpack, passive compilation sounded pretty nice to me. Letting me decide what I do is valuable, but this section makes a clear point on why it might be a bad idea (though reducing dependencies remains more important). Good section. Is there any case when using passive compilation makes sense?

Reply:
Maybe I need to enforce the point that is actually bad. 

Imagine having different c libraries that all have different sometimes overlapping and sometimes diverging compilation configurations and to make them all run together (instead of using a binary already compiled by the author) you need to accommodate your project to the needs of all those libraries (the example is a bit stretched since i believe there aren't this many discrepancies between compilation configurations in c, c++ and even none in rust), but in the _modern_ js world that is a nightmare.

Also, didn't get the point about reducing dependencies, the amount of dependencies remains about the same, they shift from the author of the component library to the consumer of the component library and contrary to empowering the consumer is just dragging it down. And as I discuss elsewhere, this will infect the rest of your ecosystem and will create a very blurry barrier between apps and components and will tight couple all of these.

Reply:
So, this actually does not give more control to the consumer, it shifts responsibilities

Reply:
True, true. Very clear. Regarding reinforcing the points in the main post, I think your second paragraph here might be a good addition but not really necessary.